src/ElasticSearch.php

```diff
@@ -95,6 +96,7 @@ public function set_engines_args( &$args ) {
 		simple_value_filter( 'epwr_decay', planet4_get_option( 'epwr_decay', 0.5 ) );
 		simple_value_filter( 'epwr_offset', planet4_get_option( 'epwr_offset', '365d' ) );
 
+		add_filter( 'ep_formatted_args', [ $this, 'ensure_function_score_exists' ], 18, 1 );
```

Comment:
It's not clear to me why we need to add another filter to solve this issue :thinking:

Reply:
Ah, is it because the logic inside it was previously in a filter that is not applied in initial and Ajax?

Reply:
I can include it in the next one, I just thought it's easier to see its purpose and remove eventually

Reply:
Adding it to `set_results_weight` seems more appropriate, because the function score is used to calculate the weight.

Reply:
`set_full_text_search` also uses it to change the query condition to a `must`. I can just call the ensure function at the start of those 2 if you prefer, or insert it in `set_full_text_search` only, but if we change the order of the 2 it will have an impact.  I'm clearly not found of having so many filters in succession, I just tried to keep the current style for now :shrug: